sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/logicaltypes/LogicalDecimal.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.schemas.logicaltypes;
+
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkArgument;
+
+import java.math.BigDecimal;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.values.Row;
+
+/** A LogicalType representing a Decimal type with custom precision and scale. */
+@Experimental(Experimental.Kind.SCHEMAS)
+public class LogicalDecimal extends IdenticalBaseTAndInputTLogicalType<BigDecimal> {
+  public static final String IDENTIFIER = "beam:logical_type:decimal:v1";
+  private static final Schema schema =
+      Schema.builder().addInt32Field("precision").addInt32Field("scale").build();
+  private final int precision;
+  private final int scale;
+
+  private LogicalDecimal(int precision, int scale) {
+    super(
+        IDENTIFIER,
+        Schema.FieldType.row(schema),
+        Row.withSchema(schema).addValues(precision, scale).build(),
+        Schema.FieldType.STRING);
```

Comment:
I'm a little confused about how this works. This argument indicates the base type is String, but this is an `IdenticalBaseTAndInputTLogicalType`, and `toBaseType` just returns a `BigDecimal`, which shouldn't be compatible.

It's possible this is a bug, it's just not turning up in any tests because we don't have a schema with LogicalDecimal being used in a coder. Could you add these new logical types to a test that exercises RowCoder? There are a variety of them: `SchemaCoderTest`, `RowCoderTest`, and `CloudObjectsTest` (a Dataflow specific test). Probably `RowCoderTest` would be the best place to test these types.

Reply:
Ah! This is a blunder. 
Sure, Brian. I will modify/add tests.

Reply:
@TheNeuralBit Modified `RowCoderTest.testLogicalType()` which exercises coder for the standard logical types.